File: 2454-largest-local-values-in-a-matrix/largest-local-values-in-a-matrix.java
class Solution {
    public int[][] largestLocal(int[][] grid) {
        int n = grid.length;
        int[][] maxLocal = new int[n - 2][n - 2];
        
        for (int i = 1; i < n - 1; i++) {
            for (int j = 1; j < n - 1; j++) {
                int[][] subMatrix = new int[3][3];
                for (int k = 0; k < 3; k++) {
                    for (int l = 0; l < 3; l++) {
                        subMatrix[k][l] = grid[i - 1 + k][j - 1 + l];
                    }
                }
                int max = Integer.MIN_VALUE;
                for (int[] row : subMatrix) {
                    for (int val : row) {
                        max = Math.max(max, val);
                    }
                }
                maxLocal[i - 1][j - 1] = max;
            }
        }
        
        return maxLocal;
    }
}

File: 2454-largest-local-values-in-a-matrix/largest-local-values-in-a-matrix.py
class Solution(object):
    def largestLocal(self, grid):
        n = len(grid)
        
        maxLocal = [[0] * (n - 2) for _ in range(n - 2)]
        for i in range(1, n - 1):
            for j in range(1, n - 1):
                subMatrix = [row[j - 1:j + 2] for row in grid[i - 1:i + 2]]
                maxLocal[i - 1][j - 1] = max(max(subMatrix, key=max))
        return maxLocal